src/utils/batchDiscordRequests.ts

```diff
@@ -0,0 +1,133 @@
+import JSONResponse from "./JsonResponse";
+import { addDelay, convertSecondsToMillis } from "./timeUtils";
+export const DISCORD_HEADERS = {
+  RATE_LIMIT_RESET_AFTER: "X-RateLimit-Reset-After",
+  RATE_LIMIT_REMAINING: "X-RateLimit-Remaining",
+  RETRY_AFTER: "Retry-After",
+};
+
+const MAX_RETRY = 1;
+const LIMIT_BUFFER = 0.2;
+
+interface RequestDetails {
+  retries: number;
+  request: () => Promise<Response>;
+  index: number;
+}
+interface ResponseDetails {
+  response: Response;
+  data: RequestDetails;
+}
+
+const parseRateLimitRemaining = (response: Response) => {
```

Comment:
. Potential overflow in rate limit calculation:

The code parses the Rate-Limit-Remaining header from the Discord API response and multiplies it by (1 - LIMIT_BUFFER).
This multiplication can potentially cause an overflow if the Rate-Limit-Remaining value is large.
While Javascript's Number type can handle large numbers, it's generally recommended to use safer methods for arithmetic operations, especially when dealing with potentially large values.

Reply:
The current expected value for `Rate-Limit-Remaining` is 10. As the max value javascript can store is 9007199254740991. It is highly unlikely that Discord will ever allow these many requests. 

> This multiplication can potentially cause an overflow if the Rate-Limit-Remaining value is large.

We are actually multiplying it with a fractional value, thus lowering the result. 
Eg: 10 * (1 - 0.2) = 8